core/src/main/java/de/danoeh/antennapod/core/feed/Feed.java

```diff
@@ -341,10 +341,7 @@ public boolean compareWithOther(Feed other) {
         if (other.isPaged() && !this.isPaged()) {
             return true;
         }
-        if (!TextUtils.equals(other.getNextPageLink(), this.getNextPageLink())) {
-            return true;
-        }
-        return false;
+        return !TextUtils.equals(other.getNextPageLink(), this.getNextPageLink());
```

Comment:
I think this is harder to read. It breaks the symmetry of the items that are compared and makes `nextPageLink` a special case. Same for the other compare functions.

Reply:
I think it's actually a bit easier to read that way, but alright, I'll change it back.